### PR TITLE
Add ESP32-C2 support

### PR DESCRIPTION
**Pull request by Jason2866**
## Description of Change
Adding the needed code changes to support build of the ESP32-C2. Libs for the C2 are still needed. 
-> Built with Arduino Lib Builder 
PR https://github.com/espressif/esp32-arduino-lib-builder/pull/149 for the Lib builder to support the C2

## Tests scenarios
The changes have been tested in Tasmota Arduino fork and project Tasmota is successful built and running for the C2

@me-no-dev as discussed in https://github.com/espressif/arduino-esp32/pull/8864

### CODE REVIEW

File: cores/esp32/esp32-hal-rgb-led.c
Comment (igrr) [inline cores/esp32/esp32-hal-rgb-led.c:4]: Note, since arduino-esp32 can use managed components now: we can use [led_strip](https://components.espressif.com/components/espressif/led_strip) component here, which has API-compatible implementations based on RMT and SPI. Then the feature will be still available for C2, and there would be less code to maintain.

File: platform.txt
Comment (igrr) [inline platform.txt:111]: There is no such config file in OpenOCD, because C2 doesn't have a USB_SERIAL_JTAG peripheral.

File: cores/esp32/esp32-hal-rgb-led.c
Comment (me-no-dev) [reply cores/esp32/esp32-hal-rgb-led.c:4]: If you are suggesting to replace our RGB code with led_strip, then this would be a larger item than this PR should handle, because it involves Peripheral Manager. I would suggest to leave it as-is and note in the PR that we need to add led_strip support (or figure out something else that would make sense)

Comment (Jason2866) [reply cores/esp32/esp32-hal-rgb-led.c:4]: My goal was to do as less as changes as possible. The suggested change is great imho should be done after this PR has merged.

Comment (me-no-dev) [reply cores/esp32/esp32-hal-rgb-led.c:4]: My suggestion would be to wrap the contents of the function instead. You can even `log_e` in `#else` to prompt us that this function is not implemented yet.

File: platform.txt
Comment (me-no-dev) [inline platform.txt:83]: You can remove `-DARDUINO_USB_MODE=1`, like it's done for ESP32

File: libraries/Wire/src/Wire.cpp
Comment (me-no-dev) [inline libraries/Wire/src/Wire.cpp:343]: why only this code and initializations above commented? If this does not cause issues, please leave it alone and add a note in the PR for me to implement it properly.

Comment (Jason2866) [reply libraries/Wire/src/Wire.cpp]: removed. Thx for looking into doing a real fix :-)

File: platform.txt
Comment (Jason2866) [reply platform.txt]: removed.

File: libraries/Wire/src/Wire.cpp
Comment (me-no-dev) [reply libraries/Wire/src/Wire.cpp]: I guess we just haven't had a chip without it yet :) no worries

File: cores/esp32/esp32-hal-rgb-led.c
Comment (Jason2866) [reply cores/esp32/esp32-hal-rgb-led.c:4]: added log_e

Comment (me-no-dev) [reply cores/esp32/esp32-hal-rgb-led.c:4]: Like this:
```cpp
void neopixelWrite(uint8_t pin, uint8_t red_val, uint8_t green_val, uint8_t blue_val){
#if SOC_RMT_SUPPORTED
    // ... the current code
#else
    log_e("RMT is not supported on " CONFIG_IDF_TARGET);
#endif /* SOC_RMT_SUPPORTED */
}
```

File: libraries/Wire/src/Wire.cpp
Comment (me-no-dev) [reply libraries/Wire/src/Wire.cpp]: https://github.com/espressif/arduino-esp32/pull/8882 created. Will merge it after CI and pull master here.

**Comment by me-no-dev** [comment]
Thanks @Jason2866 :) I'll take over to finish support

File: cores/esp32/esp32-hal-spi.c
Comment (TD-er) [inline cores/esp32/esp32-hal-spi.c:543]: Wouldn't it make much more sense to have 1 define right above where this is used to either define it as `misc` or `pin` ?

**Review by Jason2866 — COMMENTED** [review]
Ah, sorry forgot about this one. I added the missing in IDF

**Comment by me-no-dev** [comment]
@Jason2866 now it compiles as component. Will also look at the lib-builder PR

@P-R-O-C-H-Y @SuGlider @lucasssvaz PTAL

**Comment by me-no-dev** [comment]
Board Test is expected to fail due to missing SDK for C2.

**Comment by Jason2866** [comment]
@me-no-dev Thx, for the polishing and fixing.

**Comment by me-no-dev** [comment]
@Jason2866 all is good now :) both lib-builder and arduino are ready. Sketches are compiling, but we will not have it through the board manager for now. The size of the libs is 100MB. If there is enough interest or we can have them as separate package, we can include them too

**Comment by Jason2866** [comment]
@me-no-dev Big thx for the work you put in and the lightning speed!
Still curious why it is so big. Our complete framework for Arduino 3.0.0 including all MCUs and libs is around 150MB (stripped Rainmaker, SR and DSP).
The working framework for the solo1 is about 30MB https://github.com/tasmota/arduino-esp32/releases
So i expected that the C2 does add not more than 30MB.

**Comment by me-no-dev** [comment]
it adds 30MB to the ZIP and 100MB extracted. Besides that RainMaker is currently not working on it. There is close to zero interest in providing Board Manager support for it and it also takes time in the cron job that we can not afford at the time. As I said a few times, we can look into adding full support if there is enough interest or when we can make it painless to others (solo support will probably also come then)

**Comment by Jason2866** [comment]
Perfectly fine! And the size is the same the 30MB mentioned was zipped. We will see if users are asking for. I don't think Rainmaker will ever work well with the C2. The free heap is probably not enough. The C2 is not designed for heavy lifting, but it is a nice powerfull little MCU. For me it is an follow up of the ESP8266...

**Comment by me-no-dev** [comment]
>  I don't think Rainmaker will ever work well with the C2.

it's mentioned on the product page, so it should work. I'll report the bug

File: cores/esp32/esp32-hal-rgb-led.c
Comment (igrr) [reply cores/esp32/esp32-hal-rgb-led.c:3]: Yes, this comment wasn't intended to be a PR blocker. I just asked to consider this and create a follow-up task if you think this is a good idea to consider.

Comment (me-no-dev) [reply cores/esp32/esp32-hal-rgb-led.c:3]: One of the main reasons Peripheral Manager was implemented is to keep track of used peripherals and allow us to pick an available one that can achieve the requested functionality. Ex. PWM can be done with LedC, RMT, etc. Same can be said for neopixels (RMT, I2S, SPI, etc.). Probably proper task would be to implement such functionality and expand the options for all chips

Comment (Jason2866) [reply cores/esp32/esp32-hal-rgb-led.c:3]: For sure there should be a follow up here, to provide an universal working solution for every MCU, choosen by default the "best" solution (with user override, when set).

**Review by me-no-dev — APPROVED** [review]

File: variants/esp32c2/pins_arduino.h
Comment (P-R-O-C-H-Y) [inline variants/esp32c2/pins_arduino.h:21]: Let SPI pins be the same as C3 has:
```suggestion
static const uint8_t SS    = 7;
static const uint8_t MOSI  = 6;
static const uint8_t MISO  = 5;
static const uint8_t SCK   = 4;
```

**Review by P-R-O-C-H-Y — APPROVED** [review]
Please match the SPI pins with the C3. 
Tested Wifi + SD-SPI and both works well.
Thanks @Jason2866 for PR!

File: cores/esp32/HardwareSerial.h
Comment (lucasssvaz) [inline cores/esp32/HardwareSerial.h:175]: Pin 9 is already used for I2C. Maybe we can set Serial1 default pins to 10 and 18 (that are free).

```suggestion
    #elif CONFIG_IDF_TARGET_ESP32C2
      #define TX1 (gpio_num_t)18
```

File: cores/esp32/HardwareSerial.h
Comment (lucasssvaz) [inline cores/esp32/HardwareSerial.h:157]: Pin 9 is already used for I2C. Maybe we can set Serial1 default pins to 10 and 18 (that are free).

```suggestion
    #elif CONFIG_IDF_TARGET_ESP32C2
      #define RX1 (gpio_num_t)10
```

**Review by lucasssvaz — APPROVED** [review]
LGTM. I only recommend switching the default Serial1 pins to avoid possible conflicts.

**Comment by P-R-O-C-H-Y** [comment]
@Jason2866 Can you please change the Serial1 pins as suggested by @lucasssvaz? We will merge after that. Thanks

**Comment by Jason2866** [comment]
@P-R-O-C-H-Y  Changed.

**Pull request merged**
